docs/user_guide/assessment/error_estimation.rst

```diff
@@ -0,0 +1,358 @@
+    >>> # Overall fraction correct
+    >>> p = 0.5
+    >>> # Fraction of group 'A' as proportion of whole
+    >>> f_A = 0.6
+    >>> # Fraction of group 'A' which are correct
+    >>> p_A = 0.8
+
+From these, we can compute a number of other
+quantities.
+These will be used later.
+
+.. doctest:: error_estimation_code
+    :options:  +NORMALIZE_WHITESPACE
+
+    >>> # Fraction of group 'B' as proportion of whole
+    >>> f_B = 1 - f_A
+    >>> print(f_B)
+    0.4
+
+    >>> # Calculate numbers of each
+    >>> n_A = int(f_A * n_samples)
+    >>> n_B = int(f_B * n_samples)
+    >>> assert n_A + n_B == n_samples
+    >>> print(f"n_A={n_A}, n_B={n_B}")
+    n_A=600, n_B=400
+
+    >>> # Absolute number of group 'A' which are correct
+    >>> n_A_correct = int(p_A * n_A)
+    >>> print(f"n_A_correct={n_A_correct}")
+    n_A_correct=480
+
+    >>> # And for group 'B'
+    >>> n_B_correct = int(n_samples * p) - n_A_correct
+    >>> p_B = n_B_correct / n_B
+    >>> print(f"n_B_correct={n_B_correct}")
+    n_B_correct=20
+
+    >>> # Absolute numbers which are incorrect
+    >>> n_A_incorrect = n_A - n_A_correct
+    >>> n_B_incorrect = n_B - n_B_correct
+    >>> print(f"n_A_incorrect={n_A_incorrect}")
+    n_A_incorrect=120
+
+    >>> print(f"n_B_incorrect={n_B_incorrect}")
+    n_B_incorrect=380
+
+    >>> # And one more pair of sanity checks
+    >>> assert n_A == n_A_correct + n_A_incorrect
+    >>> assert n_B == n_B_correct + n_B_incorrect
+
+For simplicity, we will use 'sum' as our metric.
+Since this only requires one vector of values, but
+:class:`MetricFrame` requires metrics have the
+signature :code:`f(y_true, y_pred)`, we create a
+wrapper.
+We will ignore :code:`y_pred` in this case:
+
+
+.. doctest:: error_estimation_code
+    :options:  +NORMALIZE_WHITESPACE
+
+    >>> import numpy as np
+    >>> def bootstrap_sum(y_true, y_pred):
+    ...     return np.sum(y_true)
+
+Now, we generate the data itself.
+We need a column of 0s and 1s for the 'true' values
+(which we will also supply to :class:`MetricFrame`
+as the 'predicted' values - our metic will ignore them),
+and a column of labels 'A' and 'B' for the sensitive
+feature:
+
+.. doctest:: error_estimation_code
+    :options:  +NORMALIZE_WHITESPACE
+
+    >>> y_true = np.concatenate(
+    ...     (np.ones(int(n_samples * p)), np.zeros(int(n_samples * (1 - p))))
+    ... )
+    >>> s_f = np.concatenate(
+    ...     (
+    ...         np.full(n_A_correct, "A"),
+    ...         np.full(n_B_correct, "B"),
+    ...         np.full(n_A_incorrect, "A"),
+    ...         np.full(n_B_incorrect, "B"),
+    ...     )
+    ... )
+    
+    >>> # Show the sum
+    >>> bootstrap_sum(y_true, y_true)
+    500.0
+
+Activating Bootstrapping in :code:`MetricFrame`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+With our data in place, we can create our :class:`MetricFrame`
+object.
+In order to use bootstrapping, we specify the
+:code:`n_bootstrap_samples` and :code:`bootstrap_random_state`
```

Comment:
why does it need to be specified why the random_state is passed? On the API, we can simply have a `random_state` and use it wherever we need randomness.

Reply:
It was concern about the user passing an instance of the old or new numpy random API - and that we can't be sure that the user has a version of numpy installed which has the new API (which makes testing for types tricky. There's probably a way around this.

This comes into play when setting up the PRNGs, where each bootstrap iteration expects to get a fresh RNG, so if a single item (and not a list) is passed, I instead use the provided RNG to seed a new RNG for each iteration.

This can probably all be resolved, but for simplicity I imposed this restriction at first.

Reply:
As for which RNG, we certainly should simply support both, and also support a simple integer (as a seed).

But I was referring more to the name of the argument: `random_state` instead of `bootstrap_random_state`.

Reply:
I misunderstood your question :-) I'm pretty agnostic on the name, but I'm sure @MiroDudik will have thoughts :-)

Reply:
I agree with @adrinjalali we can just call it `random_state` (and mention in the API docs that it's only used for bootstrapping). This is also in line with the API / docs in scikit learn, e.g., [KFold](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.KFold.html#sklearn.model_selection.KFold).

Reply:
I'd be in favor of `n_boot` (following seaborn) and `random_state` (following sklearn and pandas conventions)